.github/workflows/uploadBinaries.yml

```diff
@@ -133,6 +101,7 @@ jobs:
           shasum -a 256 circt-bin-${{ matrix.runner }}.tar.gz
       - name: Upload Binaries
         uses: AButler/upload-release-assets@v2.0
+        if: startsWith(github.ref, 'refs/tags/')
```

Comment:
Nice! This is the missing thing to test this safely. 👍 

In a follow-up, this can likely be rolled into the build-and-test workflow. I.e., the build and test is the same, but the upload is gated on being on a tag.

Reply:
Absolutely.  There was mention of making re-usable helpers/functions (whatever terminology) that we could invoke from various CI places/actions, for this sort of reuse which might help in a number of places to reduce duplication / avoid drift in configs / so on.

Having these two aligned sounds great to me!  We don't currently test on OSX, but build it for that platform here, which maybe also would be good to have.  So I don't forget, cc #4473 re:considering OSX .